### PR TITLE
baseapp-auth: signup update

### PR DESCRIPTION
**Pull request by sundayguru**
* Add ExtraFields to allauth signup
* Update change email flow to create allauth email
* Implement expire password flow to allauth

### CODE REVIEW

File: baseapp-auth/baseapp_auth/extensions/allauth/account/signup.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# from allauth.account.forms import SignupForm
+from django import forms
+
+
+class ExtraSignupFormFields(forms.Form):
```
Comment (patsluth-tsl) [inline baseapp-auth/baseapp_auth/extensions/allauth/account/signup.py:5]: Can we change this file location to `baseapp-auth/baseapp_auth/extensions/allauth/account/forms.py` and change the class name to `SignupForm`? To match allAuth's file structure.

Ex: `from allauth.account.forms import SignupForm`

File: baseapp-auth/baseapp_auth/extensions/allauth/headless/adapter.py [hunk 2]
```diff
@@ -1,10 +1,28 @@
...
-        return super(HeadlessAdapter, self).serialize_user(user)
+        if user.password_expiry_date and user.password_expiry_date <= timezone.now():
+            self.request.user = user
+            token = ChangeExpiredPasswordTokenGenerator().make_token(self.request.user)
+            url = settings.FRONT_CHANGE_EXPIRED_PASSWORD_URL.format(token=token)
+            serializer = LoginChangeExpiredPasswordRedirectSerializer(
+                data=dict(redirect_url=url), context={"request": self.request}
+            )
+            serializer.is_valid(raise_exception=True)
+            return serializer.data
```
Comment (patsluth-tsl) [inline baseapp-auth/baseapp_auth/extensions/allauth/headless/adapter.py:25]: We have a follow up story for this. https://silverlogic.atlassian.net/browse/BA-1919

File: baseapp-auth/baseapp_auth/extensions/allauth/headless/adapter.py [hunk 3]
```diff
@@ -1,10 +1,28 @@
...
+        data = super(HeadlessAdapter, self).serialize_user(user)
+        serializer = UserSerializer(user, context={"request": self.request})
```
Comment (patsluth-tsl) [inline baseapp-auth/baseapp_auth/extensions/allauth/headless/adapter.py:27]: We need to be careful here. If you look at the documentation for `allauth.headless.adapter.DefaultHeadlessAdapter.serialize_user` it says

```
Returns the basic user data. Note that this data is also exposed in
        partly authenticated scenario's (e.g. password reset, email
        verification).
```

Are you sure the UserSerializer doesn't expose any unnecessary information? I believe before now we have used it only to serialize fully authenticated users.

File: baseapp-auth/baseapp_auth/rest_framework/change_email/serializers.py [hunk 4]
```diff
@@ -52,4 +53,8 @@ def update(self, instance, validated_data):
         instance.is_new_email_confirmed = False
         instance.is_email_verified = True
         instance.save()
+        EmailAddress.objects.filter(user=instance, primary=True).update(primary=False)
+        EmailAddress.objects.create(
+            user=instance, email=instance.new_email, verified=True, primary=True
+        )
```
Comment (patsluth-tsl) [inline baseapp-auth/baseapp_auth/rest_framework/change_email/serializers.py:59]: Is this a fix for https://silverlogic.atlassian.net/browse/BA-1923? Why not do it in that story?

Comment (sundayguru) [reply baseapp-auth/baseapp_auth/rest_framework/change_email/serializers.py:59]: I added in this ticket three stories just to keep the review cycle short

File: baseapp-auth/baseapp_auth/extensions/allauth/headless/adapter.py (hunk 3 above)
Comment (sundayguru) [reply baseapp-auth/baseapp_auth/extensions/allauth/headless/adapter.py:27]: Will check this, if necessary will create another serializer with what we need

File: baseapp-auth/baseapp_auth/extensions/allauth/headless/adapter.py (hunk 2 above)
Comment (sundayguru) [reply baseapp-auth/baseapp_auth/extensions/allauth/headless/adapter.py:25]: yeah, added it here.

**Comment by sundayguru** [comment]
I moved all implementation into Pat's branch

**Pull request closed without merging**